// @arcgis/map-components/dist/cjs/arcgis-table-list.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.5
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-e95448d5.js');
const index$2 = require('./index-5cfcbc22.js');
const index$3 = require('./index-04f8523d.js');
const useWidget = require('./useWidget-caab170f.js');
const index$1 = require('./index-f2bbe6e8.js');
require('./component-utils-e4944926.js');

const tableListCss = "arcgis-table-list{}";
const ArcgisTableListStyle0 = tableListCss;

const useTableListWidget = useWidget.makeWidgetController(index$3.newWidgetsTableList);
const ArcgisTableList = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.widgetTriggerAction = index.createEvent(this, "arcgisTriggerAction", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useTableListWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.collapsed = this.widget.collapsed;
        this.dragEnabled = this.widget.dragEnabled;
        this.filterPlaceholder = this.widget.filterPlaceholder;
        this.filterText = this.widget.filterText;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.minDragEnabledItems = this.widget.minDragEnabledItems;
        this.minFilterItems = this.widget.minFilterItems;
        this.selectedItems = this.widget.selectedItems;
        this.selectionMode = this.widget.selectionMode;
        this.tableItems = index$1.getSet(undefined, {
            get: () => this.widget.tableItems,
            set: "ignore",
        });
        this.listItemCreatedFunction = this.widget.listItemCreatedFunction;
        this.visibleElementsCloseButton = false;
        this.visibleElementsCollapseButton = false;
        this.visibleElementsErrors = false;
        this.visibleElementsFilter = false;
        this.visibleElementsFlow = true;
        this.visibleElementsHeading = false;
        this.visibleElementsStatusIndicators = true;
        this.visibleElementsTemporaryTableIndicators = false;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisTableList.style = ArcgisTableListStyle0;

exports.arcgis_table_list = ArcgisTableList;
