// @arcgis/map-components/dist/cjs/arcgis-weather.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.5
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-e95448d5.js');
const index$3 = require('./index-04f8523d.js');
const index$2 = require('./index-5cfcbc22.js');
const useWidget = require('./useWidget-caab170f.js');
const index$1 = require('./index-f2bbe6e8.js');
require('./component-utils-e4944926.js');

const weatherCss = "";
const ArcgisWeatherStyle0 = weatherCss;

const useWeatherWidget = useWidget.makeWidgetController(index$3.newWidgetsWeather);
const ArcgisWeather = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = useWeatherWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.hideHeader = this.widget.visibleElements.header;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisWeather.style = ArcgisWeatherStyle0;

exports.arcgis_weather = ArcgisWeather;
