// @arcgis/map-components/dist/cjs/arcgis-print.cjs.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.5
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-e95448d5.js');
const index$3 = require('./index-04f8523d.js');
const index$2 = require('./index-5cfcbc22.js');
const useWidget = require('./useWidget-caab170f.js');
const index$1 = require('./index-f2bbe6e8.js');
require('./component-utils-e4944926.js');

const printCss = "";
const ArcgisPrintStyle0 = printCss;

const usePrintWidget = useWidget.makeWidgetController(index$3.newWidgetsPrint);
const ArcgisPrint = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.arcgisReady = index.createEvent(this, "arcgisReady", 7);
        this.widgetComplete = index.createEvent(this, "arcgisComplete", 7);
        this.widgetSubmit = index.createEvent(this, "arcgisSubmit", 7);
        this.arcgisPropertyChange = index.createEvent(this, "arcgisPropertyChange", 7);
        this.manager = index$1.useControllerManager(this);
        this.widget = usePrintWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = index$2.usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.allowedFormats = index$1.getSet(this.widget.allowedFormats, {
            set: (value) => (typeof value === "string" && value !== "all" ? value.split(",") : value),
        });
        this.allowedLayouts = index$1.getSet(this.widget.allowedLayouts, {
            set: (value) => (typeof value === "string" && value !== "all" ? value.split(",") : value),
        });
        this.exportedLinks = this.widget.exportedLinks;
        this.extraParameters = this.widget.extraParameters;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.includeDefaultTemplatesDisabled = this.widget.includeDefaultTemplates;
        this.portal = this.widget.portal;
        this.printServiceUrl = this.widget.printServiceUrl;
        this.showPrintAreaEnabled = this.widget.showPrintAreaEnabled;
        this.templateCustomTextElements = this.widget.templateCustomTextElements;
        this.templateOptions = this.widget.templateOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return index.getElement(this); }
};
ArcgisPrint.style = ArcgisPrintStyle0;

exports.arcgis_print = ArcgisPrint;
