// @arcgis/map-components/dist/esm/arcgis-directional-pad.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.5
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-65c81b5d.js';
import { A as newWidgetsDirectionalPad } from './index-7838e72e.js';
import { u as usePropertyChange } from './index-877bad04.js';
import { m as makeWidgetController } from './useWidget-d88a1dfa.js';
import { u as useControllerManager } from './index-fe1504ea.js';
import './component-utils-b19c1de2.js';

const directionalPadCss = "arcgis-directional-pad{}";
const ArcgisDirectionalPadStyle0 = directionalPadCss;

const useDirectionalPadWidget = makeWidgetController(newWidgetsDirectionalPad);
const ArcgisDirectionalPad = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useDirectionalPadWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.disabled = this.widget.disabled;
        this.referenceElement = undefined;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.visualScale = this.widget.visualScale;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisDirectionalPad.style = ArcgisDirectionalPadStyle0;

export { ArcgisDirectionalPad as arcgis_directional_pad };
