// @arcgis/map-components/dist/esm/arcgis-print.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.5
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-65c81b5d.js';
import { a2 as newWidgetsPrint } from './index-7838e72e.js';
import { u as usePropertyChange } from './index-877bad04.js';
import { m as makeWidgetController } from './useWidget-d88a1dfa.js';
import { u as useControllerManager, g as getSet } from './index-fe1504ea.js';
import './component-utils-b19c1de2.js';

const printCss = "";
const ArcgisPrintStyle0 = printCss;

const usePrintWidget = makeWidgetController(newWidgetsPrint);
const ArcgisPrint = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.widgetComplete = createEvent(this, "arcgisComplete", 7);
        this.widgetSubmit = createEvent(this, "arcgisSubmit", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = usePrintWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("state");
        this.autoDestroyDisabled = false;
        this.referenceElement = undefined;
        this.allowedFormats = getSet(this.widget.allowedFormats, {
            set: (value) => (typeof value === "string" && value !== "all" ? value.split(",") : value),
        });
        this.allowedLayouts = getSet(this.widget.allowedLayouts, {
            set: (value) => (typeof value === "string" && value !== "all" ? value.split(",") : value),
        });
        this.exportedLinks = this.widget.exportedLinks;
        this.extraParameters = this.widget.extraParameters;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.headingLevel = this.widget.headingLevel;
        this.includeDefaultTemplatesDisabled = this.widget.includeDefaultTemplates;
        this.portal = this.widget.portal;
        this.printServiceUrl = this.widget.printServiceUrl;
        this.showPrintAreaEnabled = this.widget.showPrintAreaEnabled;
        this.templateCustomTextElements = this.widget.templateCustomTextElements;
        this.templateOptions = this.widget.templateOptions;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Methods
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    get el() { return getElement(this); }
};
ArcgisPrint.style = ArcgisPrintStyle0;

export { ArcgisPrint as arcgis_print };
