// @arcgis/map-components/dist/esm/arcgis-navigation-toggle.entry.js
/*!
 * All material copyright Esri, All Rights Reserved, unless otherwise specified.
 * See https://js.arcgis.com/4.31/esri/copyright.txt for details.
 * v4.31.5
 */
import { r as registerInstance, c as createEvent, g as getElement } from './index-65c81b5d.js';
import { $ as newWidgetsNavigationToggle } from './index-7838e72e.js';
import { u as usePropertyChange } from './index-877bad04.js';
import { m as makeWidgetController } from './useWidget-d88a1dfa.js';
import { u as useControllerManager } from './index-fe1504ea.js';
import './component-utils-b19c1de2.js';

const navigationToggleCss = "arcgis-navigation-toggle{}";
const ArcgisNavigationToggleStyle0 = navigationToggleCss;

const useNavigationToggleWidget = makeWidgetController(newWidgetsNavigationToggle);
const ArcgisNavigationToggle = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.arcgisReady = createEvent(this, "arcgisReady", 7);
        this.arcgisPropertyChange = createEvent(this, "arcgisPropertyChange", 7);
        this.manager = useControllerManager(this);
        this.widget = useNavigationToggleWidget(this);
        this.viewModel = this.widget.viewModel;
        this.arcgisPropertyChange = usePropertyChange()("layout", "state");
        this.autoDestroyDisabled = false;
        this.icon = this.widget.icon;
        this.label = this.widget.label;
        this.referenceElement = undefined;
        this.layout = this.widget.layout;
        this.position = "bottom-left";
        this.state = this.viewModel.state;
    }
    //--------------------------------------------------------------------------
    //
    //  Watchers
    //
    //--------------------------------------------------------------------------
    /** Permanently destroy the component */
    async destroy() {
        await this.manager.destroy();
    }
    async toggle() {
        this.widget?.toggle();
    }
    get el() { return getElement(this); }
};
ArcgisNavigationToggle.style = ArcgisNavigationToggleStyle0;

export { ArcgisNavigationToggle as arcgis_navigation_toggle };
